core-bundle/src/Command/MigrateCommand.php

```diff
@@ -0,0 +1,278 @@
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure(): void
+    {
+        $this
+            ->setName('contao:migrate')
+            ->addOption('complete', null, InputOption::VALUE_NONE, 'Execute all database migrations including DROP queries. Can be used together with --no-interaction.')
+            ->addOption('schema-only', null, InputOption::VALUE_NONE, 'Execute database schema migration only.')
+            ->setDescription('Executes migrations and the database schema diff.')
+        ;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $this->io = new SymfonyStyle($input, $output);
+
+        if ($input->getOption('schema-only')) {
+            return $this->executeSchemaDiff($input->getOption('complete')) ? 0 : 1;
+        }
+
+        if (!$this->executeMigrations()) {
+            return 1;
+        }
+
+        if (!$this->executeSchemaDiff($input->getOption('complete'))) {
+            return 1;
+        }
+
+        if (!$this->executeMigrations()) {
+            return 1;
+        }
+
+        $this->io->success('All migrations completed.');
+
+        return 0;
+    }
+
+    private function executeMigrations(): bool
+    {
+        while (true) {
+            $first = true;
+
+            foreach ($this->migrations->getPendingMigrations() as $migration) {
+                if ($first) {
+                    $this->io->section('Pending migrations');
+                    $first = false;
+                }
+
+                $this->io->writeln(' * '.$migration);
+            }
+
+            $runOnceFiles = $this->getRunOnceFiles();
+
+            if ($runOnceFiles) {
+                @trigger_error('Using runonce files has been deprecated and will no longer work in Contao 5.0. Use the migration framework instead.', E_USER_DEPRECATED);
+            }
+
+            foreach ($runOnceFiles as $file) {
+                if ($first) {
+                    $this->io->section('Pending migrations');
+                    $first = false;
+                }
+
+                $this->io->writeln(' * Runonce file: '.$file);
+            }
+
+            if ($first) {
+                break;
+            }
+
+            if (!$this->io->confirm('Execute the listed migrations?')) {
+                return false;
+            }
+
+            $this->io->section('Execute migrations');
+
+            foreach ($this->migrations->runMigrations() as $result) {
+                $this->io->writeln(' * '.$result->getMessage());
+
+                if (!$result->isSuccessful()) {
+                    $this->io->error('Migration failed');
+                }
+            }
+
+            foreach ($this->getRunOnceFiles() as $file) {
+                $this->executeRunonceFile($file);
+                $this->io->writeln(' * Executed runonce file: '.$file);
+            }
+
+            $this->io->success('Executed migrations.');
```

Comment:
This does not sound like a confirmation to me. Should it be *Executed X migrations* (with the number) or something like *All migrations executed successfully.* ?

Reply:
Agree, I will add the count.

Reply:
Done in f81820a0efa747eda89ad2b7ee98faf603a58640